spec/services/reporting/account_reuse_report_spec.rb

```diff
@@ -151,112 +151,141 @@
           created_timestamp: in_query,
           sp: all_agency_apps,
-          sp_timestamp: Array.new(12) { in_query } },
-        { id: 16, # 11 apps, 2 agencies
+          sp_timestamp: Array.new(12) { in_query },
+        },
+        { # 11 apps, 2 agencies
           created_timestamp: in_query,
           sp: all_agency_apps.first(11),
-          sp_timestamp: Array.new(11) { in_query } },
-        { id: 17, # 11 apps, 2 agencies
+          sp_timestamp: Array.new(11) { in_query },
+        },
+        { # 11 apps, 2 agencies
           created_timestamp: in_query,
           sp: all_agency_apps.first(11),
-          sp_timestamp: Array.new(11) { in_query } },
-        { id: 18, # 10 apps, 2 agencies
+          sp_timestamp: Array.new(11) { in_query },
+        },
+        { # 10 apps, 2 agencies
           created_timestamp: in_query,
           sp: all_agency_apps.first(10),
-          sp_timestamp: Array.new(10) { in_query } },
-        { id: 19, # 10 apps, 2 agencies
+          sp_timestamp: Array.new(10) { in_query },
+        },
+        { # 10 apps, 2 agencies
           created_timestamp: in_query,
           sp: all_agency_apps.first(10),
-          sp_timestamp: Array.new(10) { in_query } },
-        { id: 20, # 10 apps, 2 agencies
+          sp_timestamp: Array.new(10) { in_query },
+        },
+        { # 10 apps, 2 agencies
           created_timestamp: in_query,
           sp: all_agency_apps.first(10),
-          sp_timestamp: Array.new(9) { in_query } + Array.new(1) { out_of_query } },
-
+          sp_timestamp: Array.new(9) { in_query } + Array.new(1) { out_of_query },
+        },
+        { # 3 apps, 2 agencies - Last user gets deleted
+          created_timestamp: in_query - 1.day, # change so last is unique
+          sp: all_agency_apps.first(3),
+          sp_timestamp: Array.new(3) { in_query },
+        },
       ]
 
-      users_to_query.each do |user|
-        user[:sp].each_with_index do |sp, i|
+      mock_users_to_query.each do |mock_user|
+        current_user = create(:user, :fully_registered, registered_at: in_query)
+
+        mock_user[:sp].each_with_index do |sp, i|
           ServiceProviderIdentity.create(
-            user_id: user[:id],
+            user_id: current_user[:id],
             service_provider: sp,
-            created_at: user[:created_timestamp],
+            created_at: mock_user[:created_timestamp],
             last_ial2_authenticated_at: in_query,
-            verified_at: user[:sp_timestamp][i],
+            verified_at: mock_user[:sp_timestamp][i],
           )
         end
-      end
 
-      # Create active profiles for total_proofed_identities
-      # These 20 profiles will yield 10 active profiles in the results
-      10.times do
-        create(
-          :profile,
-          :active,
-          activated_at: in_query,
-          user: create(:user, :fully_registered, registered_at: in_query),
-        )
+        if mock_user == mock_users_to_query.last
```

Comment:
I am having a hard time articulating why but I think that this is a confusing way to handle a deleted user

Reply:
Do you mean picking the last from the array?  Or how I delete the user?  (Because I took the code for deleting a user from how users are actually deleted...)

Reply:
I meant picking the last from an array. The actual delete code is fine, sorry that was unclear

Reply:
This has been refactored after pairing on it.